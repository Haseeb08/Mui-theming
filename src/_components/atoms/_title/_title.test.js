import React from "react";
import { create } from "react-test-renderer";
import { render } from "@testing-library/react";
import Title from "./_title";

describe("Title field testing", () => {
  test("Matches the snapshot", () => {
    const title = create(<Title />);
    expect(title.toJSON()).toMatchSnapshot();
  });
  test("title field should be in document", () => {
    const { getByTestId } = render(<Title />);
    expect(getByTestId("title")).toBeInTheDocument();
  });
  test("should be have default value, if prop is not provided", () => {
    const { getByTestId } = render(<Title />);
    expect(getByTestId("title")).toHaveTextContent("My Calculator");
  });
  test("should be have input value from props", () => {
    const { getByTestId } = render(<Title title="abc" />);
    expect(getByTestId("title")).toHaveTextContent("abc");
  });
});
