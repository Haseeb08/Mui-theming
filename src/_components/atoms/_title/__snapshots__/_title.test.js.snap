// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title field testing Matches the snapshot 1`] = `
<h3
  className="MuiTypography-root MuiTypography-h3"
  data-testid="title"
>
  My Calculator
</h3>
`;
